### DISCUSSION TITLE
Movement: Stamina-Limited Movement Actions

### DISCUSSION THREAD

**Discussion by wail** [OpenTournament/OpenTournament#201] [Ideas]
This is a discussion topic related to OT-#147 issue (now closed as we have a discussion topic here).

**High level concept:**

* Players possess Stamina Meter
* Movement actions deplete player Stamina
* Movement-chaining actions may require players to manage Stamina

**e.g.**

_Each wall-dodge action may deplete player stamina, creating an inherent limitation on number of wall-dodges that can subsequently be performed._

This suggestion implies existence of movement techniques that may not currently exist, such as Dodge Jump, Wall Jump, Ledge Grab, Slide, etc.

Please discuss.

**Comment by MonsOlympus** [comment]
Ive always been a supporter of this idea myself going back to UT4s development, I believe its a good way to give people more movement options whilst mitigating abuse of aerial moves. Modern shooters such as Doom Eternal and Shadow Warrior 2 have dodge mechanics which are far more fluid but they are not suited to a Tournament environment because there needs to be some skill past spamming the movement ability. 

Managing another resource could be problematic but some talk has been in the direction of toning down some of that which could make it easier to manage another in their place. Going back to Doom Eternal it had an audio cue for when you can do these moves, it makes it easy to know in the heat of combat that option is available to you. Stamina could add another layer of skill, making a sort of rock/paper/scissors between health/armour/stamina and which you decide to sacrifice or collect at any one time and stamina can also be a solution to mitigating falling damage.

To expand on the suggestion which was made though an idea was put forward that it doesnt limit you attempting the move but limits the recovery from the move. As an easy example dodging with stamina would allow you to flow into the next move instead of having a velocity limiter on landing, the velocity limiter could be in place when dodging with no stamina so you have what is essentially a tired state. What this will do is make inputs feel better, you wont get those *donk* moments where input is non responsive, another example might be trying to wall run with low stamina limits you to one or two steps before falling where more stamina lets you hold the run. Other movements could be limited in a similar way so players know what theyve done wrong from a feedback standpoint, I think this is key to making this implementation work.
